### ISSUE TITLE
Clarify documentation of tmpfiles overrides

### ISSUE THREAD

**Issue by nh2** [systemd/systemd#15675]
[`man tmpfiles.d`](https://www.freedesktop.org/software/systemd/man/tmpfiles.d.html) says:

> All configuration files are sorted by their filename in lexicographic order, regardless of which of the directories they reside in. **If multiple files specify the same path, the entry in the file with the lexicographically earliest name will be applied.**

However, if I have to files

* `a.conf` containing `q! /tmp 1777 root root 10d`
* `b.conf` containing `q /tmp 1777 root root 10d`

then from the documentation I would expect that only `q! /tmp` applies (because they specify the same path, and `a` is lexicographically before `b`).

However, unexpectedly, they both run:

```
$ SYSTEMD_LOG_LEVEL=debug systemd-tmpfiles --clean
...
Ignoring entry q! "/tmp" because --boot is not specified.
...
Running clean action for entry q /tmp
...
```

So it seems the docs do not describe what happens.

**Comment by nh2** [comment]
Side note:

It is somewhat cumbersome that you cannot `grep` the output of `SYSTEMD_LOG_LEVEL=debug systemd-tmpfiles --clean 2>&1` to search for results. It seems that it changes its behaviour when piped, but that seems undocumented and impossible to turn off.

(Please let me know if there is a way.)

**Comment by flokli** [comment]
I think the problem here is that the override behaviour of different types isn't entirely clear, and some types make sense to be specified in conjunction, like

```
      "d '/var/spool/nginx' 0750 foouser foogrp - -"
      "Z '/var/spool/nginx' - foouser foogrp - -"
```

**Comment by poettering** [comment]
Fix waiting in #15769
